aesara/graph/basic.py

```diff
@@ -1679,8 +1679,9 @@ def expand(r):
 
             return res
 
-    results = ()
+    results: Tuple[Variable, ...] = ()
     for var in walk(graphs, expand, False):
+        assert isinstance(var, Variable)
```

Comment:
The `walk` return type depends on the third argument, but `mypy` doesn't know this.
Either this `assert` or a `not isinstance(var, Variable): raise` tells `mypy` that henceforth `var: Variable` is correct.

Reply:
Can't we use cast?

https://docs.python.org/3/library/typing.html#typing.cast

Reply:
Sound like a valid alternative: I don't have a preference.